mautic/mautic
# [UI/UX] Search (almost) Everything

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🟢 
| New feature/enhancement? (use the a.x branch)      | 🟢
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🔴
| Automated tests included?              | 🔴 
| Related user documentation PR URL      | mautic/user-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #14348 



## Description

this pr fixes mentioned issue

improves global search interface:
- adds available/active/sending, pending, expired and deactivated/unavailable statuses
- adds visual hierarchy to info
- adds icons to aid in understanding stats like read count
- shows if a searched asset is an image, video, powerpoint etc
- becomes beautiful
- makes me want to start using it. do you?
- expands the publishbadge template to support a simplified mode, where it gets smaller and no animation





| Before                                 | After
| -------------------------------------- | ---
|    ![image](https://github.com/user-attachments/assets/e595f1e3-6c40-4760-bb77-2a335edee4bf)                                     | ![image](https://github.com/user-attachments/assets/9d0b6386-5952-48a9-8228-d521705e33dd)



---
### 📋 Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
1.1. Have asset, contact, user and form created (at least)
2. Click on the global search icon on top navbar
3. Search for the stuff you have and see (or search `and` to see results for all existing things in your install; this is a bug from the functionality mentioned in #14350)



---

## The original wireframe had more functionalities
My first idea would also have **actions** appearing, specifically all buttons that were added using addButton on the LOCATION_PAGE_ACTIONS and links to all pages from the main sidebar and those from the admin dropdown (both registered in PHP too).

I'd need help to make it happen, as my tries were unsuccessful.

![Group 16](https://github.com/user-attachments/assets/17d9bdff-841a-4be4-90d3-0b9539fd266f)
